front/app/component-library/hooks/useBreakpoint.ts

```diff
@@ -11,23 +11,22 @@ const mediaQueries = {
 type BreakpointType = keyof typeof mediaQueries;
 
 const useBreakpoint = (breakpointType: BreakpointType) => {
-  const [breakpoint, setBreakpoint] = useState(
-    window.matchMedia(mediaQueries[breakpointType]).matches
+  const mediaQueryList = window.matchMedia(mediaQueries[breakpointType]);
+  const [matchesBreakpoint, setMatchesBreakpoint] = useState(
+    mediaQueryList.matches
   );
 
   useEffect(() => {
-    const media = window.matchMedia(mediaQueries[breakpointType]);
-    if (media.matches !== breakpoint) {
```

Comment:
This condition is always false since both are always equal. Since we use `breakpoint` for the if condition, we have it as a dependency. This means the effect will be recreated each time the breakpoint changes and we set it in the state. 

I didn't measure and don't believe the performance improvement will be huge, but note that this hook gets sometimes called 50-60 times on a page. So it's one of the small crumbles that add up to a pile of inefficiency that eventually results in slowness.